### PR TITLE
Fix(components): [table-column] tree checkbox selected status

### LINKED ISSUE #10199: [Component] [table-column] table 翻页记住复选框，涉及树状表格，翻页后回来，发现子项选中项会丢失选中状态，但实际选中打印集存在，只是勾选状态丢失
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.20`
- Element Plus Version: `2.2.16`
- Browser / OS: `99.0.4844.51`
- Build Tool: `Vite`

### Related Component

- `el-table-column`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFibGVcbiAgICByZWY9XCJtdWx0aXBsZVRhYmxlUmVmXCJcbiAgICA6ZGF0YT1cInRhYmxlRGF0YVwiXG4gICAgc3R5bGU9XCJ3aWR0aDogMTAwJVwiXG4gICAgOnRyZWUtcHJvcHM9XCJ7IGNoaWxkcmVuOiAnY2hpbGRyZW4nLCBoYXNDaGlsZHJlbjogJ2hhc0NoaWxkcmVuJyB9XCJcbiAgICByb3cta2V5PVwiaWRcIlxuICAgIEBzZWxlY3Rpb24tY2hhbmdlPVwiaGFuZGxlU2VsZWN0aW9uQ2hhbmdlXCJcbiAgPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gdHlwZT1cInNlbGVjdGlvblwiIHdpZHRoPVwiNTVcIiA6cmVzZXJ2ZS1zZWxlY3Rpb249J3RydWUnIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBsYWJlbD1cIkRhdGVcIiB3aWR0aD1cIjE0MFwiPlxuICAgICAgPHRlbXBsYXRlICNkZWZhdWx0PVwic2NvcGVcIj57eyBzY29wZS5yb3cuZGF0ZSB9fTwvdGVtcGxhdGU+XG4gICAgPC9lbC10YWJsZS1jb2x1bW4+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wZXJ0eT1cIm5hbWVcIiBsYWJlbD1cIk5hbWVcIiB3aWR0aD1cIjEyMFwiIC8+XG4gICAgPGVsLXRhYmxlLWNvbHVtbiBwcm9wZXJ0eT1cImFkZHJlc3NcIiBsYWJlbD1cIkFkZHJlc3NcIiBzaG93LW92ZXJmbG93LXRvb2x0aXAgLz5cbiAgPC9lbC10YWJsZT5cbiAgPGVsLXBhZ2luYXRpb24gbGF5b3V0PVwicHJldiwgcGFnZXIsIG5leHRcIiA6dG90YWw9XCIyMFwiIEBjdXJyZW50LWNoYW5nZT1cImhhbmRsZUN1cnJlbnRDaGFuZ2VcIi8+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgRWxUYWJsZSB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuaW50ZXJmYWNlIFVzZXIge1xuICBkYXRlOiBzdHJpbmdcbiAgbmFtZTogc3RyaW5nXG4gIGFkZHJlc3M6IHN0cmluZ1xuICBoYXNDaGlsZHJlbj86IGJvb2xlYW5cbiAgY2hpbGRyZW4/OiBVc2VyW11cbn1cblxuY29uc3QgbXVsdGlwbGVUYWJsZVJlZiA9IHJlZjxJbnN0YW5jZVR5cGU8dHlwZW9mIEVsVGFibGU+PigpXG5jb25zdCBtdWx0aXBsZVNlbGVjdGlvbiA9IHJlZjxVc2VyW10+KFtdKVxuY29uc3QgdG9nZ2xlU2VsZWN0aW9uID0gKHJvd3M/OiBVc2VyW10pID0+IHtcbiAgaWYgKHJvd3MpIHtcbiAgICByb3dzLmZvckVhY2goKHJvdykgPT4ge1xuICAgICAgLy8gVE9ETzogaW1wcm92ZW1lbnQgdHlwaW5nIHdoZW4gcmVmYWN0b3IgdGFibGVcbiAgICAgIC8vIGVzbGludC1kaXNhYmxlLW5leHQtbGluZSBAdHlwZXNjcmlwdC1lc2xpbnQvYmFuLXRzLWNvbW1lbnRcbiAgICAgIC8vIEB0cy1leHBlY3QtZXJyb3JcbiAgICAgIG11bHRpcGxlVGFibGVSZWYudmFsdWUhLnRvZ2dsZVJvd1NlbGVjdGlvbihyb3csIHVuZGVmaW5lZClcbiAgICB9KVxuICB9IGVsc2Uge1xuICAgIG11bHRpcGxlVGFibGVSZWYudmFsdWUhLmNsZWFyU2VsZWN0aW9uKClcbiAgfVxufVxuY29uc3QgaGFuZGxlU2VsZWN0aW9uQ2hhbmdlID0gKHZhbDogVXNlcltdKSA9PiB7XG4gIG11bHRpcGxlU2VsZWN0aW9uLnZhbHVlID0gdmFsXG59XG5cbmNvbnN0IHRhYmxlRGF0YTogVXNlcltdID0gcmVmKFtcbiAge1xuICAgIGlkOiAxLFxuICAgIGRhdGU6ICcyMDE2LTA1LTAyJyxcbiAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gIH0sXG4gIHtcbiAgICBpZDogMixcbiAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgbmFtZTogJ3dhbmd4aWFvaHUnLFxuICB9LFxuICB7XG4gICAgaWQ6IDMsXG4gICAgZGF0ZTogJzIwMTYtMDUtMDEnLFxuICAgIG5hbWU6ICd3YW5neGlhb2h1JyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBpZDogMzEsXG4gICAgICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICAgICAgbmFtZTogJ3dhbmd4aWFvaHUnLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IDMyLFxuICAgICAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgICAgIG5hbWU6ICd3YW5neGlhb2h1JyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGlkOiA0LFxuICAgIGRhdGU6ICcyMDE2LTA1LTAzJyxcbiAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gIH0sXG5dKVxuICBcbmNvbnN0IGhhbmRsZUN1cnJlbnRDaGFuZ2U9KHZhbCk9PntcbiAgY29uc29sZS5sb2codmFsKVxuICBpZih2YWw9PT0yKXtcbiAgICB0YWJsZURhdGEudmFsdWU9W1xuICAgICAge1xuICAgICAgICBpZDogNyxcbiAgICAgICAgZGF0ZTogJzIwMTgtMDUtMDInLFxuICAgICAgICBuYW1lOiAnMTIzJyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiA4LFxuICAgICAgICBkYXRlOiAnMjAxNi0wNS0wNCcsXG4gICAgICAgIG5hbWU6ICczMzMnLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgaWQ6IDksXG4gICAgICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICAgICAgbmFtZTogJzMyMjIzJyxcbiAgICAgICAgY2hpbGRyZW46IFtcbiAgICAgICAgICB7XG4gICAgICAgICAgICBpZDogMjIsXG4gICAgICAgICAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgICAgICAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gICAgICAgICAgfSxcbiAgICAgICAgICB7XG4gICAgICAgICAgICBpZDogMTIsXG4gICAgICAgICAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgICAgICAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gICAgICAgICAgfSxcbiAgICAgICAgXSxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIGlkOiAxMCxcbiAgICAgICAgZGF0ZTogJzIwMTYtMDUtMDMnLFxuICAgICAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gICAgICB9LFxuXVxuICAgIFxuICB9ZWxzZXtcbiAgICB0YWJsZURhdGEudmFsdWU9W1xuICB7XG4gICAgaWQ6IDEsXG4gICAgZGF0ZTogJzIwMTYtMDUtMDInLFxuICAgIG5hbWU6ICd3YW5neGlhb2h1JyxcbiAgfSxcbiAge1xuICAgIGlkOiAyLFxuICAgIGRhdGU6ICcyMDE2LTA1LTA0JyxcbiAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gIH0sXG4gIHtcbiAgICBpZDogMyxcbiAgICBkYXRlOiAnMjAxNi0wNS0wMScsXG4gICAgbmFtZTogJ3dhbmd4aWFvaHUnLFxuICAgIGNoaWxkcmVuOiBbXG4gICAgICB7XG4gICAgICAgIGlkOiAzMSxcbiAgICAgICAgZGF0ZTogJzIwMTYtMDUtMDEnLFxuICAgICAgICBuYW1lOiAnd2FuZ3hpYW9odScsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBpZDogMzIsXG4gICAgICAgIGRhdGU6ICcyMDE2LTA1LTAxJyxcbiAgICAgICAgbmFtZTogJ3dhbmd4aWFvaHUnLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuICB7XG4gICAgaWQ6IDQsXG4gICAgZGF0ZTogJzIwMTYtMDUtMDMnLFxuICAgIG5hbWU6ICd3YW5neGlhb2h1JyxcbiAgfSxcbl1cbiAgfVxuICBcbiAgXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

树状表格带复选框，翻页记忆功能，来回翻页后，树状子集已勾选，打印勾选子集有，但是复选框没有勾上（即状态错误）

### What is Expected?

勾选数组有的，复选框状态希望正确

### What is actually happening?

翻页回来，子集是已勾选，但是复选框状态是未勾选

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by Simon-He95**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix: #10199

### CODE REVIEW

**Comment by chenxch** [comment]
There is a problem that the child node cannot be reversed, maybe you can add a test case.

**Comment by Simon-He95** [comment]
> There is a problem that the child node cannot be reversed, maybe you can add a test case.

I have fixed, pls check

File: packages/components/table/src/store/watcher.ts [hunk 1]
```diff
@@ -137,7 +137,7 @@ function useWatcher<T>() {
 
   // 选择
   const isSelected = (row) => {
-    return selection.value.includes(row)
+    return selection.value.some((item) => item.id === row.id)
```
Comment (tolking) [inline packages/components/table/src/store/watcher.ts:140]: Does this `id` attribute always exist?

Comment (Simon-He95) [reply packages/components/table/src/store/watcher.ts]: good question, I have replaced `id` with `row-key`

Comment (tolking) [reply packages/components/table/src/store/watcher.ts]: Keep the original code does not work properly?

**Comment by tolking** [comment]
1. You should look at the type of `row-key`.
2. Avoid unnecessary changes.
3. Some checks were not successful, please check.

**Comment by fenghua123** [comment]
When will it be resolved? It's urgent

**Comment by tolking** [comment]
There are some problems with this PR.

**Comment by fenghua123** [comment]
When will this bug be fixed？

**Comment by fenghua123** [comment]
Didn't anyone follow up on this problem

**Pull request closed without merging**